source/iterator/pub_sub_listener.go

```diff
@@ -59,18 +59,21 @@ func NewPubSubIterator(ctx context.Context, client redis.Conn, key string) (*Pub
 }
 
 // HasNext returns whether there are any more records to be returned
-func (i *PubSubIterator) HasNext(_ context.Context) bool {
+func (i *PubSubIterator) HasNext() bool {
 	return len(i.records) > 0 || !i.tomb.Alive()
```

Comment:
So, if the tomb is not alive, we say a next element exists?

Reply:
This is to ensure the proper error is returned by the Next function. Took reference from: https://github.com/ConduitIO/conduit-connector-s3/blob/main/source/iterator/cdc_iterator.go#L74

Reply:
Gotcha :+1: Let's then please leave a comment, similar to the one in the S3 connector. We can also update the comment for this method to say that Next returns true if there's a record OR an error to be returned.

Reply:
Done